pkg/lca/ibu_test.go

```diff
@@ -0,0 +1,693 @@
+		expectedErrorText   string
+	}{
+		{
+			expectedError:       nil,
+			addToRuntimeObjects: true,
+		},
+	}
+
+	for _, testCase := range testCases {
+		var (
+			runtimeObjects []runtime.Object
+			testSettings   *clients.Settings
+		)
+
+		testIBU := generateImageBasedUpgrade()
+
+		if testCase.addToRuntimeObjects {
+			runtimeObjects = append(runtimeObjects, testIBU)
+		}
+
+		testSettings = clients.GetTestClients(clients.TestClientParams{
+			K8sMockObjects: runtimeObjects,
+		})
+
+		ibuBuilder, err := PullImageBasedUpgrade(testSettings)
+		assert.Nil(t, err)
+
+		// Test the PullImageBasedUpgrade function
+		builderResult, err := ibuBuilder.WithSeedImage("quay.io/no-image").Update()
+
+		// Check the error
+		assert.Equal(t, err, testCase.expectedError)
+
+		if testCase.expectedError == nil {
+			assert.NotNil(t, builderResult)
+		}
+	}
+}
+
+func TestImageBasedUpgradeDelete(t *testing.T) {
+	testCases := []struct {
+		expectedError       error
+		addToRuntimeObjects bool
+		expectedErrorText   string
+	}{
+		{
+			expectedError:       nil,
+			addToRuntimeObjects: true,
+		},
+	}
+
+	for _, testCase := range testCases {
+		var (
+			runtimeObjects []runtime.Object
+			testSettings   *clients.Settings
+		)
+
+		testIBU := generateImageBasedUpgrade()
+
+		if testCase.addToRuntimeObjects {
+			runtimeObjects = append(runtimeObjects, testIBU)
+		}
+
+		testSettings = clients.GetTestClients(clients.TestClientParams{
+			K8sMockObjects: runtimeObjects,
+		})
+
+		ibuBuilder, err := PullImageBasedUpgrade(testSettings)
+		assert.Nil(t, err)
+
+		// Test the Delete function
+		builderResult, err := ibuBuilder.Delete()
+
+		// Check the error
+		assert.Equal(t, err, testCase.expectedError)
+
+		if testCase.expectedError == nil {
+			assert.Nil(t, builderResult.Object)
+		}
+	}
+}
+
+func TestImageBasedUpgradeGet(t *testing.T) {
+	testCases := []struct {
+		expectedError       error
+		addToRuntimeObjects bool
+		expectedErrorText   string
+	}{
+		{
+			expectedError:       nil,
+			addToRuntimeObjects: true,
+		},
+	}
+
+	for _, testCase := range testCases {
+		var (
+			runtimeObjects []runtime.Object
+			testSettings   *clients.Settings
+		)
+
+		testIBU := generateImageBasedUpgrade()
+
+		if testCase.addToRuntimeObjects {
+			runtimeObjects = append(runtimeObjects, testIBU)
+		}
+
+		testSettings = clients.GetTestClients(clients.TestClientParams{
+			K8sMockObjects: runtimeObjects,
+		})
+
+		ibuBuilder, err := PullImageBasedUpgrade(testSettings)
+		assert.Nil(t, err)
+
+		// Test the Get function
+		builderResult, err := ibuBuilder.Get()
+
+		// Check the error
+		assert.Equal(t, err, testCase.expectedError)
+
+		if testCase.expectedError == nil {
+			assert.NotNil(t, builderResult)
+		}
+	}
+}
+
+func TestImageBasedUpgradeExists(t *testing.T) {
+	testCases := []struct {
+		expectedError       error
+		addToRuntimeObjects bool
+		expectedErrorText   string
+	}{
+		{
+			expectedError:       nil,
+			addToRuntimeObjects: true,
+		},
```

Comment:
I think you should include a negative test

Reply:
Added